pkg/operation/common/utils_test.go

```diff
@@ -45,6 +45,36 @@ var _ = Describe("common", func() {
 			})
 		})
 
+		Describe("#DistributePercentOverZones", func() {
+			It("should return unmodified percentage if total is evenly divisble", func() {
+				var (
+					total      = 6
+					noOfZones  = 3
+					percentage = "40%"
+				)
+
+				percentages := make([]string, 0, noOfZones)
+				for i := 0; i < noOfZones; i++ {
+					percentages = append(percentages, ExportDistributePercentOverZones(i, percentage, noOfZones, total))
+				}
+				Expect(percentages).To(Equal([]string{percentage, percentage, percentage}))
+			})
+
+			It("should return correct percentage if total is not evenly divisble", func() {
+				var (
+					total      = 7
+					noOfZones  = 3
+					percentage = "40%"
+				)
+
+				percentages := make([]string, 0, noOfZones)
+				for i := 0; i < noOfZones; i++ {
+					percentages = append(percentages, ExportDistributePercentOverZones(i, percentage, noOfZones, total))
+				}
+				Expect(percentages).To(Equal([]string{"52%", "35%", "35%"}))
```

Comment:
Having 3 zones and 7 replicas, the `DistributeOverZones` function would assign the first 2 zones 2 replicas each and 3 replicas to the last zone. Now, when distributing 40% over 3 zones we would expect the first 2 to have a little bit less (35%) while the last one would get a little more (52%). However, your test succeeds with a different order.

Reply:
DistributeOverZones acts the other way around - The first zone will get 3 replicas and the last two zones will get 2 replicas each, thus I'd say that the weighing here should be correct.